### PR TITLE
fix: loan and items

### PR DESCRIPTION
**Pull request by rerowep**
* Corrects small/big loans and items files. There were too few items for the small files.
   * Improves error handling and logging for JSON reference resolvers.
   * Uses more reliable loading of json files for records creation.
   * Improves functions parameters documentation.

Co-Authored-by: Peter Weber <peter.weber@rero.ch>

## Why are you opening this PR?

- fixes jsonresolver errors

## How to test?

- run normal and deployment setup and check for jsonresolver errors

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: rero_ils/modules/jsonresolver.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Holding resolver."""
+
+import click
+from flask import current_app
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+
+
+def ref_resolver(pid_type, pid):
```
Comment (jma) [inline rero_ils/modules/jsonresolver.py]: Can you update your documentation with :param, etc. and change the name of the function. A class would be a resolver, a function resolve -> `def resolve_jsonreference` or something like this.

File: rero_ils/modules/jsonresolver.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            ),
+            fg='red'
+        )
+    else:
+        if persistent_id.status == PIDStatus.REGISTERED:
+            return dict(pid=persistent_id.pid_value)
+        current_app.logger.error(
+            'Doc resolver error: /api/documents/{pid} {persistent_id}'.format(
+                pid=pid,
+                persistent_id=persistent_id
```
Comment (zannkukai) [inline rero_ils/modules/jsonresolver.py]: 'documents' shouldn't be a static string, should be dynamic depending `pid_type` parameter

File: rero_ils/modules/jsonresolver.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Holding resolver."""
```
Comment (zannkukai) [inline rero_ils/modules/jsonresolver.py]: Not needed

File: rero_ils/modules/jsonresolver.py (hunk 2 above)
Comment (rerowep) [reply rero_ils/modules/jsonresolver.py]: ok

File: rero_ils/modules/jsonresolver.py (hunk 3 above)
Comment (rerowep) [reply rero_ils/modules/jsonresolver.py]: ok

File: rero_ils/modules/jsonresolver.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        )
+    else:
+        if persistent_id.status == PIDStatus.REGISTERED:
+            return dict(pid=persistent_id.pid_value)
+        current_app.logger.error(
+            '{pid_type} resolve error: /api/documents/{pid} {id}'.format(
+                pid_type=pid_type,
+                pid=pid,
+                id=persistent_id
+            )
```
Comment (zannkukai) [inline rero_ils/modules/jsonresolver.py]: 'documents' static string into URL should be dynamic.
Something like
```python
'{pid_type} resolve error: api/{item_route} {id}'.format(
  pid_type=pid_type,
  
item_route=current_app.config.get('RECORDS_REST_ENDPOINTS').get(pid_type).get('item_route').format(pid_value=pid),
  pid=pid,
  id=persistent_id
)
```
I don't know if my `item_route` suggestion works.... not yet the know_how XD

**Pull request merged**